### PR TITLE
Egress Rules Refactoring 2

### PR DESCRIPTION
**Pull request by vadimeisenbergibm**
**What this PR does / why we need it**:
Improves egress rules handling code.

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:
It is resubmission of https://github.com/istio/istio/pull/1863, rebased.

**Release note**:

```NONE
```

### CODE REVIEW

**Comment by vadimeisenbergibm** [comment]
@ijsnellf I applied your comments for #1863, in this rebased PR. 
Note that string in https://github.com/istio/istio/pull/1887/commits/117ea6ea6cb042796bf383244e0b6bdd10a9129a#diff-9d998bcb98e2ea8ae4a360f52661eaabL886 is still split since it is long.

File: pilot/model/egress_rules.go [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+	// the key here is the key of the Istio configuration objects - see
+	// `func (meta *ConfigMeta) Key() string`
+	for key := range egressRules {
+		keys = append(keys, key)
+	}
+	sort.Strings(keys)
+
+	// domains - a map where keys are of the form domain:port and values are the keys of
+
+	// domains - a map where keys are of the form domain:port and values are the keys of
```
Comment (ijsnellf) [inline pilot/model/egress_rules.go]: nit: duplicated lines

**Comment by ijsnellf** [comment]
Looks good to me, except for the nit

**Comment by vadimeisenbergibm** [comment]
@ijsnellf Could you please approve the PR and add LGTM label?

**Pull request merged**